frontend/context/StakingContractDetailsProvider.tsx

```diff
@@ -0,0 +1,181 @@
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import { AGENT_CONFIG } from '@/config/agents';
+import { GNOSIS_CHAIN_CONFIG } from '@/config/chains';
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { useService } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { StakingContractDetails } from '@/types/Autonolas';
+
+import {
+  INITIAL_DEFAULT_STAKING_PROGRAM_ID,
+  StakingProgramContext,
+} from './StakingProgramProvider';
+
+const currentAgent = AGENT_CONFIG.trader; // TODO: replace with dynamic agent selection
+const currentChainId = GNOSIS_CHAIN_CONFIG.chainId; // TODO: replace with dynamic chain selection
+
+// TODO: create a separate hook to get service id
+const useServiceId = () => {
+  const { selectedService, isFetched: isLoaded } = useServices();
+  const serviceConfigId =
+    isLoaded && selectedService ? selectedService?.service_config_id : '';
+  const { service } = useService({ serviceConfigId });
+  const serviceId = service?.chain_configs[currentChainId].chain_data?.token;
+
+  return serviceId;
+};
+
+/**
+ * hook to get staking contract details by staking program
+ */
+const useStakingContractDetailsByStakingProgram = (isPaused: boolean) => {
+  const { activeStakingProgramId } = useContext(StakingProgramContext);
+  const serviceId = useServiceId();
+
+  return useQuery({
+    queryKey: [
+      REACT_QUERY_KEYS.STAKING_CONTRACT_DETAILS_BY_STAKING_PROGRAM_KEY,
+      currentChainId,
+      serviceId,
+      activeStakingProgramId,
+    ],
+    queryFn: async () => {
+      return await currentAgent.serviceApi.getStakingContractDetailsByServiceIdStakingProgram(
+        serviceId!,
+        activeStakingProgramId!,
+        currentChainId,
+      );
+    },
+    enabled:
+      !!serviceId && !!activeStakingProgramId && !currentChainId && !isPaused,
+    refetchInterval: !isPaused ? FIVE_SECONDS_INTERVAL : false,
+    refetchOnWindowFocus: false,
+  });
+};
+
+/**
+ * hook to get all staking contract details
+ */
+const useAllStakingContractDetails = () => {
+  const stakingPrograms = Object.values([INITIAL_DEFAULT_STAKING_PROGRAM_ID]);
+
+  const queryResults = useQueries({
+    queries: stakingPrograms.map((programId) => ({
+      queryKey: REACT_QUERY_KEYS.ALL_STAKING_CONTRACT_DETAILS(
+        currentChainId,
+        programId,
+      ),
+      queryFn: async () =>
+        await currentAgent.serviceApi.getStakingContractDetailsByStakingProgram(
+          programId,
+          currentChainId,
+        ),
+      onError: (error: Error) => {
+        console.error(
+          `Error fetching staking details for ${programId}:`,
+          error,
+        );
+      },
+    })),
+  });
```

Comment:
nice work on these, but please move them to hooks and export, as they'll be useful for other components